File: src/main/java/org/mitallast/queue/transport/TransportService.java
package org.mitallast.queue.transport;

import org.mitallast.queue.common.codec.Message;

public interface TransportService {

    void connectToNode(DiscoveryNode node);

    void disconnectFromNode(DiscoveryNode node);

    void send(DiscoveryNode node, Message message);
}

File: src/main/java/org/mitallast/queue/transport/TransportService.kt
package org.mitallast.queue.transport

import org.mitallast.queue.common.codec.Message

interface TransportService {

    fun connectToNode(node: DiscoveryNode)

    fun disconnectFromNode(node: DiscoveryNode)

    fun send(node: DiscoveryNode, message: Message)
}
